sdk/core/azure-core-amqp/src/amqp/connection.cpp

```diff
@@ -506,7 +506,7 @@ namespace Azure { namespace Core { namespace Amqp { namespace _detail {
         << "ConnectionImpl::Close: " << this << " ID: " << m_containerId;
     if (!m_connection)
     {
-      throw std::logic_error("Connection already closed.");
+      throw std::logic_error("Connection not opened.");
```

Comment:
Do we know it was closed or should this be:
"Connection invalid. The connection may not be open."

Reply:
We know that the connection was not opened. We don't know why it was not opened.

The error was *not* caused because the connection was already closed, because `m_connection` is only released in the destructor for the connection.